pkg/proxy/backup.go

```diff
@@ -0,0 +1,444 @@
+package proxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
+
+	eclient "github.com/longhorn/longhorn-engine/pkg/controller/client"
+	rclient "github.com/longhorn/longhorn-engine/pkg/replica/client"
+	esync "github.com/longhorn/longhorn-engine/pkg/sync"
+	etypes "github.com/longhorn/longhorn-engine/pkg/types"
+	eutil "github.com/longhorn/longhorn-engine/pkg/util"
+	eptypes "github.com/longhorn/longhorn-engine/proto/ptypes"
+
+	"github.com/longhorn/backupstore"
+	bsutil "github.com/longhorn/backupstore/util"
+)
+
+func (p *Proxy) SnapshotBackup(ctx context.Context, req *rpc.EngineSnapshotBackupRequest) (resp *rpc.EngineSnapshotBackupProxyResponse, err error) {
+	log := logrus.WithFields(logrus.Fields{"serviceURL": req.ProxyEngineRequest.Address})
+	log.Debugf("Backing up snapshots %v to backup %v", req.SnapshotName, req.BackupName)
+
+	resp = &rpc.EngineSnapshotBackupProxyResponse{
+		ProxyEngineResponse: &rpc.ProxyEngineResponse{},
+		BackupId:            "",
+		Replica:             "",
+		IsIncremental:       false,
+	}
+	ackResponse(resp.ProxyEngineResponse, false)
+
+	newCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Why not just use `ctx` instead of creating another `newCtx`? the same question to other methods.

Reply:
Fixed.